packages/appsync-emulator-serverless/vtl.js

```diff
@@ -57,6 +57,15 @@ class JavaString {
     return new JavaString(rep);
   }
 
+  split(regexString, limit = undefined) {
+    const re = new RegExp(
+      typeof regexString === 'object' ? regexString.strVal : regexString,
+    );
+    const result = this.strVal.split(re, limit);
+    // java.util.String.split does not seem to include the regex in the result ...
+    return result.filter((v, ii) => (ii + 1) % 2).map(v => new JavaString(v));
```

Comment:
Two concerns here:

1) We are assuming the javascript regexp implementation to be the same as the java regexp implementation ( which I am okay with merging, but we should add a comment and maybe something to the README about this assumption )

2) Its unclear to me why we are filtering the split result:

```
>let result = '1a2b3c4d5e6f'.split(new RegExp(/\d/))
[ '', 'a', 'b', 'c', 'd', 'e', 'f' ]
> result.filter((v, ii) => (ii + 1) % 2)
[ '', 'b', 'd', 'f' ]
```

Reply:
1.According to [wikipedia](https://en.wikipedia.org/wiki/Comparison_of_regular_expression_engines#Language_features) at least implemented features are almost identical (no lookback in JS). I'll add a warning.
2. What I was trying to implement is that [Java](https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#split(java.lang.String)) does not return the separator, ie `"abc".split("(b)|(d)") == ["a", "b"] `. However just realized while writing this up that JS is more [complicated](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split#Description) than I though. The capturing groups (not the full match) get spliced into the returned array. My test works b/c I got one capturing group, your's does not because there is none. I'll provide a fix that works with an arbitrary number of capturing groups

Reply:
Sounds good 👍